### ISSUE TITLE
Methods annotated with @UnknownInitialization(This.class) may NPE without error

### ISSUE THREAD

**Issue by mihai-dumitresq** [typetools/checker-framework#905]
Here's the failing test case, using v2.1.1:
```Java
import org.checkerframework.checker.initialization.qual.UnknownInitialization;
class This {
  private final Object mBar;
  This() {
    baz();
    mBar = "";
  }

  void baz(@UnknownInitialization(This.class) This this) {
    mBar.toString();
  }

  public static void main(String[] args) {
    new This();
  }
}
```

Also, to avoid introducing false positives, can this test case also be added when fixing this issue?

```Java
import org.checkerframework.checker.initialization.qual.UnknownInitialization;
class This {
  private final Object mBar;
  This() {
    mBar = "";
    // Should be fine, This is now fully initialized.
    baz();
  }

  void baz(@UnknownInitialization(This.class) This this) {
    mBar.toString();
  }
}
```

I wanted to double check my understanding of @UnknownInitialization so wrote what should have been error producing code. I was following the docs here:
http://types.cs.washington.edu/checker-framework/current/checker-framework-manual.html#becoming-initialized-clarification

**Comment by wmdietl** [comment]
Thanks for that report!
I've added a test case in `checker/tests/nullness/init/Issue905.java`.
The call of baz in the first constructor should be forbidden, as the object is not yet initialized up-to and including `This`.
The second invocation should work, as that frame has been initialized.
Before closing this issue we should add the second test case to ensure no false positives are added.
